### PR TITLE
Allow optional space for param detection

### PR DESCRIPTION
**Pull request by RyanCopley**
Really, I intend for this to be more of a discussion than anything. Now that I think I understand what's going on, i can proceed with my work, but want to make sure this is known.

I work for a company that has a large argo workflow, and we've been struggling to reconcile strange behaviors around the space needed for the interpolated variables. I believe this is because this specific regex is not meeting the contract that was documented, and does not allow for those additional white space between the bracket and variable.

https://github.com/argoproj/argo-workflows/blob/master/docs/variables.md#simple

> Simple tags may have white-space between the brackets and variable.

https://github.com/argoproj/argo-workflows/blob/master/util/template/simple_template.go#L13 provides that contract during the workflows actual execution (e.g. `argo submit`), however, there is a code path during `argo lint` that tries to use the file i'm changing to determine if it is or isn't a parameter, in order to detect if it should allow that as a "global" param, but it doesn't follow that same contract?

https://github.com/argoproj/argo-workflows/blob/master/workflow/validate/validate.go#L872

I posted about this in the slack:

https://cloud-native.slack.com/archives/C01QW9QSSSK/p1653497167145119

https://cloud-native.slack.com/archives/C01QW9QSSSK/p1653600495901169

### CODE REVIEW

**Comment by tczhao** [comment]
related https://github.com/argoproj/argo-workflows/issues/4286

**Review by tczhao — CHANGES_REQUESTED** [review]
A few things:

We'd better move this to an issue if its intended for discussion
For the PR. Can you fix the DCO, see [link](https://github.com/argoproj/argo-workflows/pull/8867/checks?check_run_id=6616955074) or `git commit -s --amend`

There seems more cases where this white space thing is not respected, e.g. https://github.com/argoproj/argo-workflows/blob/master/workflow/validate/validate.go#L1270. I'd probably start with searching all the `{{` codes...

We need to add more test to validate these edge cases

File: workflow/validate/validate.go [hunk 1]
```diff
@@ -1373,7 +1373,7 @@ func sortDAGTasks(tmpl *wfv1.Template, ctx *dagValidationContext) error {
 
 var (
 	// paramRegex matches a parameter. e.g. {{inputs.parameters.blah}}
-	paramRegex               = regexp.MustCompile(`{{[-a-zA-Z0-9]+(\.[-a-zA-Z0-9_]+)*}}`)
```
Comment (sarabala1979) [inline workflow/validate/validate.go:1376]: Can you add  test?

**Comment by RyanCopley** [comment]
I don't intend to come across as rude, but unfortunately my schedule doesn't allow me to put more effort into this when, now that I did this analysis, we have a way forward avoiding this problem. That means I have no interest (read: time, motivation) in any further commits to this PR to bring it up to your standards. I simply wish to let you all know that this contract is being broken, and a very thin example of what the issue is so you can understand and prioritize it on your own.

Good luck though, this issue was incredibly hard to wrap my head around and has led to toil on my team :(  For the record, we'll be standardizing on wether or not to allow spaces in these tags, and update our existing workflows accordingly.

**Comment by alexec** [comment]
No more work expected on this PR. Closing.

**Comment by agilgur5** [comment]
I believe this was superseded by #11781 and perhaps some PRs before it
